routers/api/v1/org/team.go

```diff
@@ -215,14 +226,17 @@ func EditTeam(ctx *context.APIContext, form api.EditTeamOption) {
 	}
 
 	if team.Authorize < models.AccessModeOwner {
-		var units = make([]*models.TeamUnit, 0, len(form.Units))
-		for _, tp := range unitTypes {
-			units = append(units, &models.TeamUnit{
-				OrgID: ctx.Org.Team.OrgID,
-				Type:  tp,
-			})
+		if len(form.Units) > 0 {
+			var units = make([]*models.TeamUnit, 0, len(form.Units))
+			unitTypes := models.FindUnitTypes(form.Units...)
+			for _, tp := range unitTypes {
+				units = append(units, &models.TeamUnit{
+					OrgID: ctx.Org.Team.OrgID,
+					Type:  tp,
+				})
+			}
+			team.Units = units
 		}
-		team.Units = units
 	}
 
 	if err := models.UpdateTeam(team, isAuthChanged, isIncludeAllChanged); err != nil {
```

Comment:
But this function will update all fields of the struct?

Reply:
@lunny yes it does --- thats how the function wors NOW